### ISSUE TITLE
Memory Leak in the function mbedtls_rsa_rsassa_pkcs1_v15_sign

### ISSUE THREAD

**Issue by JayaraghavendranK** [ARMmbed/mbedtls#372]
In the function mbedtls_rsa_rsassa_pkcs1_v15_sign, there is a possibility of memory leak in the below code segment (Line No 1088 for mbed-tls 2.2.0 version):

```C
    sig_try = mbedtls_calloc( 1, ctx->len );
    verif   = mbedtls_calloc( 1, ctx->len );
    if( sig_try == NULL || verif == NULL )
        return( MBEDTLS_ERR_MPI_ALLOC_FAILED );
```

In this case, if memory allocation to sig_try succeeds, but, if memory allocation to verif fails, then sig_try pointer is not freed which can lead to memory leak.

Simple Fix Suggestion:

```C
    sig_try = mbedtls_calloc( 1, ctx->len );
    if(NULL == sig_try)
        return( MBEDTLS_ERR_MPI_ALLOC_FAILED );
    verif   = mbedtls_calloc( 1, ctx->len );
    if( NULL == verif)
    {
        mbedtls_free(sig_try);
        return( MBEDTLS_ERR_MPI_ALLOC_FAILED );
    }
```

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTSSL-579

**Comment by sbutcher-arm** [comment]
This issue has been fixed in mbed TLS releases, 2.2.1, 2.1.4 and 1.3.16.
